src/core/lib/security/credentials/plugin/plugin_credentials.h

```diff
@@ -21,10 +21,22 @@
 
 #include "src/core/lib/security/credentials/credentials.h"
 
-typedef struct {
+struct grpc_plugin_credentials;
+
+typedef struct grpc_plugin_credentials_pending_request {
+  bool cancelled;
+  struct grpc_plugin_credentials *creds;
+  grpc_credentials_mdelem_list *md_list;
+  grpc_closure *on_request_metadata;
+  struct grpc_plugin_credentials_pending_request *prev;
```

Comment:
Making it a double-linked list makes it more efficient to remove the pending request from the list when it is successfully completed, which is the common case (see plugin_credentials.c lines 70-72 -- if this wasn't a double-linked list, we would need to iterate through the list in order to remove an element).  I think it's particularly important to make this case fast, because the work has to be done while holding the credential's mutex.  So if there are multiple calls in flight at once (which should be common), then we will be blocking progress on all other calls while we're removing one call from the pending request list.